### PR TITLE
Fixes #23318 - Add available_for to errata API

### PR DESCRIPTION
**Pull request by PaulSD**
See https://github.com/Katello/katello/pull/7324#issuecomment-386642181

### CODE REVIEW

**Comment by jlsherrill** [comment]
[test katello]

File: app/controllers/katello/api/v2/errata_controller.rb [hunk 1]
```diff
@@ -3,21 +3,28 @@ class Api::V2::ErrataController < Api::V2::ApiController
...
+    param :available_for, String, :desc => N_("Return errata that can be added to the specified object.  The values 'content_view_version' and 'content_view_filter are supported.")
     param_group :search, Api::V2::ApiController
     def index
-      params[:errata_restrict_applicable] = false if ::Foreman::Cast.to_bool(params[:errata_restrict_installable])
       super
     end
 
+    def available_for_content_view_version(version)
+      version.available_errata
+    end
```
Comment (jlsherrill) [inline app/controllers/katello/api/v2/errata_controller.rb]: I don't see this method used anywhere, nor do i see the available_for=content_view_version actually working properly.

Comment (PaulSD) [reply app/controllers/katello/api/v2/errata_controller.rb]: It is called from https://github.com/Katello/katello/pull/7324/commits/1c9cfb8d87fe2441d5ca1b22be18a38e543d92b6#diff-afb34388d8bbd87b85d8adbd9d0b2cb1R69

I started reworking this PR per https://github.com/Katello/katello/pull/7324#issuecomment-386642181 but figured I should wait for #7324 to be merged before finishing this up, so this was incomplete.

I'll rebase this on the current master shortly.

**Comment by PaulSD** [comment]
These test failures aren't my fault :)

**Comment by jlsherrill** [comment]
Should be fixed with a [test katello] :)

**Pull request merged**